pkg/extension/extension_test.go

```diff
@@ -0,0 +1,100 @@
+package extension_test
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	eirinixcatalog "code.cloudfoundry.org/eirinix/testing"
+	. "github.com/SUSE/eirini-dns-aliases/pkg/extension"
+	"github.com/docker/libnetwork/resolvconf"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func jsonifyPatches(resp admission.Response) []string {
+	var r []string
+	for _, patch := range resp.Patches {
+		r = append(r, patch.Json())
+	}
+	return r
+}
+
+func decodePatches(resp admission.Response) string {
+	return strings.Join(jsonifyPatches(resp), "")
+}
+
+const (
+	noDnsPolicyPatch = `{"op":"add","path":"/spec/dnsPolicy","value":"None"}`
+)
+
+var _ = Describe("Eirini extension", func() {
+	eirinixcat := eirinixcatalog.NewCatalog()
+	extension := NewExtension("", "")
+	eiriniManager := eirinixcat.SimpleManager()
+	request := admission.Request{}
+	pod := &corev1.Pod{}
+
+	JustBeforeEach(func() {
+		extension = NewExtension("", "")
+		eirinixcat = eirinixcatalog.NewCatalog()
+		eiriniManager = eirinixcat.SimpleManager()
+
+		raw, err := json.Marshal(pod)
+		Expect(err).ToNot(HaveOccurred())
+
+		request = admission.Request{AdmissionRequest: admissionv1beta1.AdmissionRequest{Object: runtime.RawExtension{Raw: raw}}}
+	})
+
+	Describe("eirini-dns-aliases", func() {
+		Context("when handling a Eirini runtime app", func() {
+			BeforeEach(func() {
+				pod = &corev1.Pod{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name: "opi",
+							},
+						},
+					},
+				}
+			})
+
+			It("Does patch the dns policy", func() {
+				rc, err := resolvconf.Get()
+				Expect(err).ToNot(HaveOccurred())
+
+				searches := resolvconf.GetSearchDomains(rc.Content)
+				searches = append(searches, "service.cf.internal")
+
+				patches := jsonifyPatches(extension.Handle(context.TODO(), eiriniManager, pod, request))
```

Comment:
```suggestion
				patches := jsonifyPatches(extension.Handle(context.Background(), eiriniManager, pod, request))
```
The docs for [`Background`](https://pkg.go.dev/context#Background) explicitly mention using it in tests, so it should be the correct one. (`TODO` is more as a placeholder until we finish writing the code that would generate the correct context.)